discord/sku.py

```diff
@@ -0,0 +1,138 @@
+
+if TYPE_CHECKING:
+    from datetime import datetime
+
+    from .guild import Guild
+    from .state import ConnectionState
+    from .types.sku import (
+        SKU as SKUPayload,
+        Entitlement as EntitlementPayload,
+    )
+
+__all__ = (
+    'SKU',
+    'Entitlement',
+)
+
+
+class SKU:
+    """Represents a premium offering as a stock-keeping unit (SKU).
+
+    .. versionadded:: 2.4
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The SKU's ID.
+    type: :class:`SKUType`
+        The type of the SKU.
+    application_id: :class:`int`
+        The ID of the application that the SKU belongs to.
+    name: :class:`str`
+        The consumer-facing name of the premium offering.
+    slug: :class:`str`
+        A system-generated URL slug based on the SKU name.
+    """
+
+    __slots__ = (
+        '_state',
+        'id',
+        'type',
+        'application_id',
+        'name',
+        'slug',
+        '_flags',
+    )
+
+    def __init__(self, *, state: ConnectionState, data: SKUPayload):
+        self._state: ConnectionState = state
+        self.id: int = int(data['id'])
+        self.type: SKUType = try_enum(SKUType, data['type'])
+        self.application_id: int = int(data['application_id'])
+        self.name: str = data['name']
+        self.slug: str = data['slug']
+        self._flags: int = data['flags']
+
+    def __repr__(self) -> str:
+        return f'<SKU id={self.id} name={self.name!r} slug={self.slug!r}>'
+
+    @property
+    def flags(self) -> SKUFlags:
+        """Returns the flags of the SKU."""
+        return SKUFlags._from_value(self._flags)
+
+
+class Entitlement:
+    """Represents an entitlement from user or guild which has been granted access to a premium offering.
+
+    .. versionadded:: 2.4
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The entitlement's ID.
+    sku_id: :class:`int`
+        The ID of the SKU that the entitlement belongs to.
+    user_id: :class:`int`
+        The ID of the user that is granted access to the entitlement.
+    guild_id: Optional[:class:`int`]
+        The ID of the guild that is granted access to the entitlement
+    application_id: :class:`int`
+        The ID of the application that the entitlement belongs to.
+    type: :class:`EntitlementType`
+        The type of the entitlement.
+    consumed: :class:`bool`
+        Whether the entitlement has been consumed.
+    starts_at: Optional[:class:`datetime.datetime`]
+        A UTC start date which the entitlement is valid. Not present when using test entitlements.
+    ends_at: Optional[:class:`datetime.datetime`]
+        A UTC date which entitlement is no longer valid. Not present when using test entitlements.
+    deleted: :class:`bool`
+        Whether the entitlement has been deleted. Not applicable to app subscriptions, since they are not consumable.
+    subscription_id: Optional[:class:`int`]
+        The ID of the subscription that the entitlement belongs to if any.
```

Comment:
These two fields are undocumented. They are in the examples though, so it's probably okay.

Reply:
I think they can be removed since they add no value. `deleted` should be always false, since when the sku is deleted, is should just disappear.

Reply:
Talked with Mason. Those attributes won't provide any value